src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityAlarm.java

```diff
@@ -0,0 +1,152 @@
+import gregtech.api.metatileentity.TieredMetaTileEntity;
+import gregtech.api.metatileentity.interfaces.IGregTechTileEntity;
+import gregtech.api.terminal.gui.widgets.SelectorWidget;
+import gregtech.client.renderer.texture.Textures;
+import gregtech.core.sound.GTSoundEvents;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundEvent;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MetaTileEntityAlarm extends TieredMetaTileEntity {
+    private SoundEvent selectedSound;
+    private boolean isActive;
+    public static final int BASE_EU_CONSUMPTION = 4;
+
+    public MetaTileEntityAlarm(ResourceLocation metaTileEntityId) {
+        super(metaTileEntityId, 1);
+        selectedSound = GTSoundEvents.DEFAULT_ALARM;
+    }
+
+    @Override
+    public MetaTileEntity createMetaTileEntity(IGregTechTileEntity tileEntity) {
+        return new MetaTileEntityAlarm(metaTileEntityId);
+    }
+
+    @Override
+    public void addInformation(ItemStack stack, @Nullable World player, @NotNull List<String> tooltip, boolean advanced) {
+        super.addInformation(stack, player, tooltip, advanced);
+        tooltip.add(I18n.format("gregtech.machine.alarm.tooltip"));
+        tooltip.add(I18n.format("gregtech.universal.tooltip.uses_per_tick", BASE_EU_CONSUMPTION));
+        tooltip.add(I18n.format("gregtech.universal.tooltip.energy_storage_capacity", energyContainer.getEnergyCapacity()));
+    }
+
+    public void renderMetaTileEntity(CCRenderState renderState, Matrix4 translation, IVertexOperation[] pipeline) {
+        super.renderMetaTileEntity(renderState, translation, pipeline);
+        Textures.ALARM_OVERLAY.renderSided(getFrontFacing(), renderState, translation, pipeline);
+    }
+
+    @Override
+    protected ModularUI createUI(EntityPlayer entityPlayer) {
+        return ModularUI.builder(GuiTextures.BACKGROUND, 240, 50)
+                .widget(new LabelWidget(5, 5, getMetaFullName()))
+                .widget(new SelectorWidget(10, 20, 220, 20,
+                        getSounds().stream().map((event) -> event.getSoundName().toString()).collect(Collectors.toList()),
+                        0x555555, () -> this.selectedSound.getSoundName().toString(), true).setOnChanged((v) -> {
+                    GregTechAPI.soundManager.stopTileSound(getPos());
+                    this.selectedSound = SoundEvent.REGISTRY.getObject(new ResourceLocation(v));
+                    this.writeCustomData(GregtechDataCodes.UPDATE_SOUND, (writer) -> writer.writeResourceLocation(this.selectedSound.getSoundName()));
```

Comment:
This should not be done if the selected sound is the same as the sound that is currently playing, because you didn't really change the sound.